app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -205,9 +205,29 @@ def miq_templates
       operating_system.product_name = guest_os
       operating_system.distribution = i.try(:os_distro)
       operating_system.version = i.try(:os_version)
+
+      if snapshot?(i) && parent_server_uid
+        snapshot = persister.snapshots.find_or_build(i.id)
+        snapshot.name = i.name
+        snapshot.uid  = i.id
+        snapshot.uid_ems = i.id
+        snapshot.ems_ref = i.id
+        snapshot.create_time = i.created_at
+        snapshot.description = i.attributes[:description]
+        snapshot.vm_or_template = persister.vms.lazy_find(parent_server_uid)
+      end
     end
   end
 
+  def snapshot?(image)
+    block_device_mapping = image.attributes[:block_device_mapping]
+    source_type = JSON.parse(block_device_mapping).try(:first).try(:[], 'source_type') if block_device_mapping
```

Comment:
Is the `.try(:first)` necessary?  I don't think it is possible for `JSON.parse()` to not throw a parser error and also return nil.  Or is this if it returns something that doesn't respond_to `:first` ?

Reply:
Good catch, that should be removed. It's an artifact of me originally using `JSON.load` there.

Reply:
Ah okay, `&.dig("source_type")` possibly a little cleaner, idk why but the `.try(:[], 'source_type')` always looked a little strange to me

Reply:
Heh agreed, .try(:[], _) is a pretty weird construct.